From dd79edb2b8fcc300cd417259797dab52a2f5f32b Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@marvell.com>
Date: Mon, 11 Feb 2019 12:29:40 -0800
Subject: [PATCH 0435/1921] spi: octeontx2: add entry of pci device id for 95xx

Add entry for 95xx subsytem ID in pci device ID match table.

Change-Id: Id896c449343b5ef22f75dd9ae4cbc83025e41df8
Signed-off-by: Suneel Garapati <sgarapati@marvell.com>
Reported-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/3823
Reviewed-by: Peter Swain <pswain@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/8662
Reviewed-by: Suneel Garapati <sgarapati@caviumnetworks.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/26930
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/spi/spi-octeontx2.c | 9 +++++++--
 drivers/spi/spi-octeontx2.h | 4 +++-
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/drivers/spi/spi-octeontx2.c b/drivers/spi/spi-octeontx2.c
index 25dbf102f7ba..e39c4f046e4d 100644
--- a/drivers/spi/spi-octeontx2.c
+++ b/drivers/spi/spi-octeontx2.c
@@ -323,9 +323,14 @@ static void octeontx2_spi_remove(struct pci_dev *pdev)
 }
 
 static const struct pci_device_id octeontx2_spi_pci_id_table[] = {
-	{ PCI_DEVICE_SUB(PCI_VENDOR_ID_CAVIUM, 0xa00b,
+	{ PCI_DEVICE_SUB(PCI_VENDOR_ID_CAVIUM,
+			 PCI_DEVID_OCTEONTX2_SPI,
 			 PCI_VENDOR_ID_CAVIUM,
-			 PCI_SUBSYS_DEVID_OCTEONTX2) },
+			 PCI_SUBSYS_DEVID_OTX2_96XX) },
+	{ PCI_DEVICE_SUB(PCI_VENDOR_ID_CAVIUM,
+			 PCI_DEVID_OCTEONTX2_SPI,
+			 PCI_VENDOR_ID_CAVIUM,
+			 PCI_SUBSYS_DEVID_OTX2_95XX) },
 	{ 0, }
 };
 
diff --git a/drivers/spi/spi-octeontx2.h b/drivers/spi/spi-octeontx2.h
index 52611f3ba487..d7038522b555 100644
--- a/drivers/spi/spi-octeontx2.h
+++ b/drivers/spi/spi-octeontx2.h
@@ -4,7 +4,9 @@
 
 #include <linux/clk.h>
 
-#define PCI_SUBSYS_DEVID_OCTEONTX2	0xB200
+#define PCI_DEVID_OCTEONTX2_SPI 0xA00B
+#define PCI_SUBSYS_DEVID_OTX2_96XX 0xB200
+#define PCI_SUBSYS_DEVID_OTX2_95XX 0xB300
 
 #define OCTEONTX2_SPI_MAX_BYTES 1024
 #define OCTEONTX2_SPI_MAX_CLOCK_HZ 25000000
-- 
2.31.1

